eng/packaging.targets

```diff
@@ -71,35 +72,33 @@
   </Target>
 
   <Target Name="AddRuntimeSpecificAssemblies"
-          DependsOnTargets="BuiltProjectOutputGroup;SatelliteDllsProjectOutputGroup;_AddPriFileToPackBuildOutput;$(TargetsForTfmSpecificBuildOutput)"
+          DependsOnTargets="BuiltProjectOutputGroup;SatelliteDllsProjectOutputGroup;$(TargetsForTfmSpecificBuildOutput)"
           Condition="'$(TargetsAnyOS)' != 'true' and '$(ExcludeFromPackage)' == ''">
     <ItemGroup>
       <TfmSpecificPackageFile Include="@(SatelliteDllsProjectOutputGroupOutput);
-                            @(BuiltProjectOutputGroupOutput);
-                            @(_PathToPriFile)">
+                            @(BuiltProjectOutputGroupOutput)">
         <PackagePath>runtimes/$(_runtimeOS)/lib/$(TargetFrameworkWithoutSuffix)</PackagePath>
       </TfmSpecificPackageFile>
     </ItemGroup>
   </Target>
 
   <Target Name="AddRefDebugSymbolsWithTfm"
-          Condition="'$(IncludeSymbols)' == 'true' and '$(IncludeRefInPackage)' == 'true' and '$(TargetsAnyOS)' == 'true'">
-    <PropertyGroup>
-      <RefSymbolPath>$(BaseOutputPath)ref\$(TargetFrameworkWithoutSuffix)-$(Configuration)\$(TargetName).pdb</RefSymbolPath>
-    </PropertyGroup>
-
-    <ItemGroup Condition="Exists('$(RefSymbolPath)')">
-      <TfmSpecificDebugSymbolsFile Include="$(RefSymbolPath)">
-        <TargetPath>/ref/$(TargetFrameworkWithoutSuffix)/$(TargetName).pdb</TargetPath>
-        <TargetFramework>$(TargetFrameworkWithoutSuffix)</TargetFramework>
+          Condition="'$(IncludeSymbols)' == 'true' and '$(PackageReferenceAssembly)' == 'true' and '@(ResolvedMatchingContract)' != ''">
+    <ItemGroup>
+      <TfmSpecificDebugSymbolsFile Include="$([System.IO.Path]::GetDirectoryName('%(ResolvedMatchingContract.Identity)'))\$(TargetName).pdb">
+        <RefAssemblyOutput>true</RefAssemblyOutput>
+        <!-- ResolvedMatchingContract.SetTargetFramework has string value in format "TargetFramework=net6.0". We remove "TargetFramework=" to get the nearest target framework."-->
+        <TargetPath>/ref/$([System.String]::new('%(ResolvedMatchingContract.SetTargetFramework)').Substring(16))\$(TargetName).pdb</TargetPath>
+        <TargetFramework>$([System.String]::new('%(ResolvedMatchingContract.SetTargetFramework)').Substring(16))</TargetFramework>
```

Comment:
nit: can you set the TargetFramework attribute first and then reuse its value in the TargetPath to avoid the duplicate substring and string new operation?